report/simple_report.go

```diff
@@ -0,0 +1,174 @@
+package report
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const SEV_LOW = "low"
+const SEV_MEDIUM = "medium"
+const SEV_HIGH = "high"
+const SEV_CRITICAL = "critical"
+
+type Risk struct {
+	Severity    string   `json:"severity"`
+	Title       string   `json:"title"`
+	Description string   `json:"description"`
+	Fix         string   `json:"fix"`
+	Items       []string `json:"items"`
+}
+
+type SimpleReport struct {
+	HiddenService string `json:"hiddenService"`
+	Risks         []Risk `json:"risks"`
+}
+
+func (osr *SimpleReport) AddRisk(severity string, title string, description string, fix string, items []string) {
+	osr.Risks = append(osr.Risks, Risk{severity, title, description, fix, items})
+}
+
+// Format as JSON
+func (osr *SimpleReport) Serialize() (string, error) {
+	report, err := json.Marshal(osr)
+	if err != nil {
+		return "", err
+	}
+	return string(report), nil
+}
+
+var risk_levels = map[string]string{
+	SEV_LOW:      "\033[094mLow Risk:\033[0m",
+	SEV_MEDIUM:   "\033[093mMedium Risk:\033[0m",
+	SEV_HIGH:     "\033[091mHigh Risk:\033[0m",
+	SEV_CRITICAL: "\033[091mCritical Risk:\033[0m",
+}
+
+// Format as human-readable text to be printed to console
+func (osr *SimpleReport) Format() (string, error) {
+	buffer := bytes.NewBuffer(nil)
+	buffer.WriteString("--------------- OnionScan Report ---------------\n")
+
+	buffer.WriteString(fmt.Sprintf("Generating Report for: %s\n\n", osr.HiddenService))
+
+	for _, risk := range osr.Risks {
+		buffer.WriteString(risk_levels[risk.Severity] + " " + risk.Title + "\n")
+		if len(risk.Description) > 0 {
+			for _, item := range strings.Split(risk.Description, "\n") {
+				buffer.WriteString("\t " + item + "\n")
```

Comment:
Since we are here....this might be a good place to do the formatting of the output to console....rather than in the description itself - to ensure we have a consistency....I'm not sure how easy this to do though.